client/src/main/java/io/pravega/client/netty/impl/ConnectionFactory.java

```diff
@@ -24,21 +25,23 @@
     /**
      * Establishes a connection between server and client with given parameters.
      *
+     * @param id       Identifier
      * @param endpoint The Pravega Node URI.
      * @param rp       Reply Processor instance.
      * @return An instance of client connection.
      */
-    CompletableFuture<ClientConnection> establishConnection(PravegaNodeUri endpoint, ReplyProcessor rp);
+    CompletableFuture<ClientConnection> establishConnection(UUID id, PravegaNodeUri endpoint, ReplyProcessor rp);
 
     /**
      * This method is used to establish a client connection using a {@link Flow} on the underlying Connection
      * pool.
      * @param flow  Flow to be used to create a client connection.
+     * @param id     Identifier
      * @param endpoint The Pravega Node URI.
      * @param rp Reply Processor instance.
      * @return An instance of client connection.
      */
-    CompletableFuture<ClientConnection> establishConnection(Flow flow, PravegaNodeUri endpoint, ReplyProcessor rp);
+    CompletableFuture<ClientConnection> establishConnection(Flow flow, UUID id, PravegaNodeUri endpoint, ReplyProcessor rp);
```

Comment:
I don't think we need to add UUID. See my comment below.

Reply:
yes its done.. file changes are reduced from 22 to 12 now.